### ISSUE TITLE
docs: remove pin Sphinx<6.0.0 after next Celery release

### ISSUE THREAD

**Issue by GraemeWatt** [HEPData/hepdata#594]
The last two nightly CI runs ([30 December](https://github.com/HEPData/hepdata/actions/runs/3804267459) and [31 December](https://github.com/HEPData/hepdata/actions/runs/3810111163)) failed with an error message when building the docs:
```bash
TypeError: Python directive method get_signature_prefix() must return a list of nodes. Return value was '(task)'.
```
This coincides with the release of Sphinx v6.0.0.  I've fixed the docs build by pinning `Sphinx<6.0.0`.  The Celery extension for Sphinx has been fixed in celery/celery#7978, but we'll need to wait for a new Celery release before removing the pin `Sphinx<6.0.0`.

**Comment by GraemeWatt** [comment]
Celery v5.3.0 was released on 6th June 2023, but [invenio-celery](https://github.com/inveniosoftware/invenio-celery) pins `celery>=5.1.0,<5.3`, so we need to wait for a new `invenio-celery` release with the upper pin removed.

**Comment by GraemeWatt** [comment]
[invenio-celery v1.3.0](https://pypi.org/project/invenio-celery/1.3.0/) was released on 5th March 2024 with `celery>=5.1.0,<5.4` so the pin `Sphinx<6.0.0` can now be removed from the HEPData code.
